libs/designer-v2/src/lib/ui/panel/recommendation/details/connectorDetailsView.tsx

```diff
@@ -0,0 +1,151 @@
+          if (!shouldEnableNestedAgent || !(isWithinAgenticLoop || isAgentTool)) {
+            return false;
+          }
+        }
+
+        if (shouldEnableParseDocWithMetadata === false && data.id === 'parsedocumentwithmetadata') {
+          return false;
+        }
+
+        return true;
+      },
+    [isAgentTool, isRoot, isWithinAgenticLoop, shouldEnableNestedAgent, shouldEnableParseDocWithMetadata]
+  );
+
+  const allOperations: OperationActionData[] = useMemo(() => {
+    if (!groupOperations) {
+      return [];
+    }
+
+    return groupOperations
+      .map((operation) => OperationActionDataFromOperation(operation))
+      .filter(filterItems)
+      .sort((a, b) => {
+        const aTitle = a.title ?? '';
+        const bTitle = b.title ?? '';
+        return aTitle.localeCompare(bTitle);
+      });
+  }, [groupOperations, filterItems]);
+
+  const triggers = useMemo(() => allOperations.filter((op) => op.isTrigger), [allOperations]);
+  const actions = useMemo(() => allOperations.filter((op) => !op.isTrigger), [allOperations]);
+
+  const connectorName = connector?.properties?.displayName || 'Connector';
+  const connectorDescription = connector?.properties?.description;
+  const connectorIcon = connector?.properties?.iconUri;
+  const brandColor = connector?.properties?.brandColor || '#0078d4';
+
+  // Show loading state
+  if (isLoading || (!groupOperations && !isNullOrUndefined(connector))) {
+    return (
+      <div className={classes.container}>
+        <div className={classes.header}>
+          {connectorIcon && <img src={connectorIcon} alt={connectorName} className={classes.connectorIcon} />}
+          <div className={classes.headerContent}>
+            <div style={{ display: 'flex', alignItems: 'center', gap: '8px' }}>
+              <Text weight="semibold" size={400} className={classes.connectorName} style={{ color: brandColor }}>
+                {connectorName}
+              </Text>
+              <FavoriteButton connectorId={connectorId} />
+            </div>
+            {connectorDescription ? (
+              <Text size={300} className={classes.connectorDescription}>
+                {connectorDescription}
+              </Text>
+            ) : null}
+          </div>
+        </div>
+        <div style={{ display: 'flex', justifyContent: 'center', padding: '20px' }}>
+          <Spinner size="medium" label="Loading operations..." />
+        </div>
+      </div>
+    );
+  }
+
+  if (isNullOrUndefined(connector)) {
+    return null;
+  }
+
+  return (
+    <div className={classes.container}>
+      <div className={classes.header}>
+        {connectorIcon && <img src={connectorIcon} alt={connectorName} className={classes.connectorIcon} />}
+        <div className={classes.headerContent}>
+          <div style={{ display: 'flex', alignItems: 'center', gap: '8px' }}>
+            <Text weight="semibold" size={400} className={classes.connectorName} style={{ color: brandColor }}>
```

Comment:
We should have a check that the brand color is not too close to the background color, some connectors have a really light brand color (I think some just have white) and if it's too light / dark we should default to black so it's still visible